StyleCop.Analyzers/StyleCop.Analyzers/ReadabilityRules/SA1112ClosingParenthesisMustBeOnLineOfOpeningParenthesis.cs

```diff
@@ -47,7 +50,88 @@ public override ImmutableArray<DiagnosticDescriptor> SupportedDiagnostics
         /// <inheritdoc/>
         public override void Initialize(AnalysisContext context)
         {
-            // TODO: Implement analysis
+            context.RegisterSyntaxNodeAction(HandleMethodDeclaration, SyntaxKind.MethodDeclaration);
+            context.RegisterSyntaxNodeAction(HandleConstructorDeclaration, SyntaxKind.ConstructorDeclaration);
+            context.RegisterSyntaxNodeAction(HandleInvocationExpression, SyntaxKind.InvocationExpression);
+            context.RegisterSyntaxNodeAction(HandleObjectCreationExpression, SyntaxKind.ObjectCreationExpression);
         }
+
+        private void HandleObjectCreationExpression(SyntaxNodeAnalysisContext context)
+        {
+            var objectCreation = (ObjectCreationExpressionSyntax)context.Node;
+            if (objectCreation.ArgumentList.IsMissing ||
+                objectCreation.ArgumentList.Arguments.Count > 0)
+            {
+                return;
+            }
+
+            if (objectCreation.ArgumentList.OpenParenToken.IsMissing == false &&
+                    objectCreation.ArgumentList.CloseParenToken.IsMissing == false)
+                {
+                    CheckIfLocationOfOpenAndCloseTokensAreTheSame(context,
+                        objectCreation.ArgumentList.OpenParenToken, objectCreation.ArgumentList.CloseParenToken);
+                }
+        }
+
+        private void HandleInvocationExpression(SyntaxNodeAnalysisContext context)
+        {
+            var invocationExpression = (InvocationExpressionSyntax) context.Node;
+            if (invocationExpression.ArgumentList.IsMissing ||
+                invocationExpression.ArgumentList.Arguments.Count > 0)
+            {
+                return;
+            }
+
+            if (invocationExpression.ArgumentList.OpenParenToken.IsMissing == false &&
```

Comment:
I'm not sure how @sharwell sees this but ```== false``` does not really follow the coding style used on the other diagnostics.